test_type/cross_shard.js

```diff
@@ -0,0 +1,243 @@
+  }
+  return testList;
+}
+
+function addCrossShardTestJob(testList) {
+  for (const test of testList) {
+    test.jobList.push({
+      workerUrl: test.config.workerUrl,
+      input: {
+        type: JobType.CROSS_SHARD_TEST,
+        config: {
+          ...test.config,
+        },
+      },
+      output: {},
+    });
+  }
+}
+
+async function requestJob(job) {
+  if (job.status === JobStatus.PASS) {
+    return;
+  }
+
+  try {
+    const response = await request({
+      method: 'post',
+      baseURL: job.workerUrl,
+      url: '/job',
+      data: job.input,
+    });
+    job.status = JobStatus.PROGRESS;
+    job.id = response.data.id;
+  } catch (err) {
+    console.log(`Error while request job (${err.message})`);
+    job.status = JobStatus.FAIL;
+    job.output.message = err.message;
+  }
+}
+
+async function processJob(testList, jobIndex) {
+  console.log(`- Start to process '${testList[0].jobList[jobIndex].input.type}' job`);
+  const requestList = [];
+  for (const test of testList) {
+    requestList.push(requestJob(test.jobList[jobIndex]));
+  }
+  await Promise.all(requestList);
+}
+
+async function waitJob(testList, jobIndex) {
+  let unfinishedCount;
+  const jobType = testList[0].jobList[jobIndex].input.type;
+
+  do {
+    unfinishedCount = 0;
+
+    for (const test of testList) {
+      const job = test.jobList[jobIndex];
+
+      if (job.status !== JobStatus.PROGRESS) {
+        continue;
+      }
+
+      try {
+        const response = await request({
+          method: 'get',
+          baseURL: job.workerUrl,
+          url: `/job/${job.id}`,
+        });
+
+        job.status = response.data.status;
+        job.output = response.data.output;
+        if (job.status === JobStatus.PROGRESS) {
+          unfinishedCount++;
+        }
+      } catch (err) {
+        job.status = JobStatus.FAIL;
+        job.output.message = err.message;
+      }
+    }
+    if (unfinishedCount === 0) {
+      break;
+    }
+    console.log(`${unfinishedCount} workers are still processing '${jobType}' job ` +
+        `(${testList.length - unfinishedCount}/${testList.length}) [${getRunningTime()}]`);
+    await delay(10000);
+  } while (true);
+}
+
+async function processCrossShardTestJob(testList) {
+  addCrossShardTestJob(testList);
+  await processJob(testList, 0);
+  await waitJob(testList, 0);
+}
+
+function makeRoundList(testList) {
+  // Make dataPool
+  const dataPool = [];
+  for (const [index, test] of testList.entries()) {
+    if (!test.jobList[0].output.matchedList) {
+      continue;
+    }
+    for (const matchedData of test.jobList[0].output.matchedList) {
+      matchedData.shardNumber = index + 1;
+      dataPool.push(matchedData);
+    }
+  }
+  dataPool.sort((a, b) => {
+    return a.sentAt < b.sentAt ? -1 : a.sentAt > b.sentAt ? 1 : 0;
+  });
+
+  // Slice by time zone
+  const roundList = [];
+  for (const test of testList) {
+    const startTime = _.get('test.jobList[0].output.matchedList[0].sentAt', 0);
```

Comment:
Should this be `_.get(test, 'jobList[0].output.matchedList[0].sentAt', 0)` ?

Reply:
Right!!! Thanks for catching my mistake. Done

Reply:
Thanks for the quick updates! LGTM!